src/Editor.tsx

```diff
@@ -1,57 +1,82 @@
-          <div key={key}>
-            {toppings[layer].name}
-            <button
-              children="⬆️"
-              onClick={() =>
-                changeDoc(
-                  (doc) =>
-                    void doc.layers.splice(
-                      key - 1,
-                      0,
-                      doc.layers.splice(key, 1)[0]
-                    )
-                )
-              }
-            />
-            <button
-              children="⬇️"
-              onClick={() =>
-                changeDoc(
-                  (doc) =>
-                    void doc.layers.splice(
-                      key + 1,
-                      0,
-                      doc.layers.splice(key, 1)[0]
-                    )
-                )
-              }
-            />
-            <button
-              children="🗑️"
-              onClick={() => changeDoc((doc) => void doc.layers.splice(key, 1))}
-            />
-          </div>
-        ))}
-      </div>
-    </div>
-  );
+export const Editor = ({doc, changeDoc}) => {
+    return (
+        <div>
+            <hr/>
+            <div style={{padding: '1rem'}}>
+                {Object.entries(toppings).map(([id]) => (
+                    <button
+                        key={id}
+                        style={{display: "block", marginBottom: '0.3rem'}}
+                        children={`+ ${toppings[id].name}`}
+                        onClick={() => changeDoc((doc) => void doc.layers.push(id))}
+                    />
+                ))}
+            </div>
+            <hr/>
+            <div id="list" style={{minWidth: '290px'}}>
+                {doc?.layers.map((layer, key) => (
+                    <div key={key} style={{
+                        cursor: 'move',
+                        userSelect: 'none',
+                        display: 'flex',
+                        justifyContent: 'space-between',
+                        margin: '1rem',
+                        border: '1px dotted white',
+                        padding: '1rem'
+                    }}>
+                        <div>{toppings[layer].name}</div>
+
+                        <div>{key > 0 ? <button
+                            style={{marginLeft: '0.3rem'}}
+                            children="⬆️"
+                            onClick={() => {
+
+                                changeDoc(
+                                    (doc) => {
+                                        const replacedLayer = doc.layers[key - 1];
+                                        const untouched = doc.layers.slice(key + 1)
+                                        const frontLayer = [...doc.layers]
+                                        frontLayer.splice(key - 1, untouched.length + 1, layer)
+                                        frontLayer.push(...[...untouched])
+                                        frontLayer[key] = replacedLayer
+                                        doc.layers = frontLayer
```

Comment:
You have to mutate the array in-place for automerge, immutable methods ([...array]) don't provide it with the change information it needs

Reply:
I mutate as the last step

Reply:
ok, I don't have the mental capacity to do this at the moment sorry :(

Reply:
it's ok, you can look into it when you can

Reply:
but I've tested it and it works fine

Reply:
https://github.com/neftaly/autoburger/assets/11094475/b383fde8-5583-4c4e-94f1-7b62a564b508